### ISSUE TITLE
Nested methodrefs can't capture variables

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1324]
The code below fails to compile because the nested methodref can't capture `i`.

```
		[Inline]
		public static void ForEach<TList, TValue, TDlg>(this TList list, TDlg dlg)
			where TList : concrete, IEnumerable<TValue>
			where TDlg : delegate void(TValue)
		{
			for (let item in list)
				dlg(item);
		}

		public class CTest
		{
			public int Test;
			public List<CTest> List;
		}

		public static List<CTest> list;

		public static int Main(String[] args)
		{
				int i = 0;

				list.ForEach((l) => l.List.ForEach((l) =>
				{
					Console.WriteLine(i); // Identifier not found
				}));
		}
```

But if you change the code to:
```
				list.ForEach((c) =>
				{
					Console.WriteLine(i);
					c.List.ForEach((c) =>
					{
						Console.WriteLine(i);
					});
				});
```

Then it compiles. I guess this makes sense since `i` was captured by the parent methodref.

**Comment by bfiete** [comment]
Fixed at eae4aed3a969908e585a88222ee0a6a22f9965d8

**Comment by disarray2077** [comment]
@bfiete Oops...  The code in this issue seems to have hidden another issue, the issue about capturing variables was solved but for some reason, the parameter of the lambda isn't usable...

Example:
```
			list.ForEach((l) => l.List.ForEach((l2) =>
			{
				Console.WriteLine(l2.Test); // Identifier not found
			}));
```

**Comment by bfiete** [comment]
Should be fixed at af0cd194db48751fcdc3c1f718ad57a9939d05d0
